File: examples/tiktoken_drop_in.py
"""Drop gigatoken into code written for tiktoken.

`as_tiktoken()` wraps a Tokenizer in a drop-in for the `tiktoken.Encoding`
API. Run with: uv run examples/tiktoken_drop_in.py
"""

import gigatoken as gt

# Any Tokenizer source works; a .tiktoken vocabulary file loads directly
# with gt.Tokenizer.from_tiktoken("r50k_base.tiktoken").
encoding = gt.Tokenizer("openai-community/gpt2").as_tiktoken()

ids = encoding.encode("Tokenize your text data at GB/s!")
print(ids)
print(encoding.decode(ids))

# Special-token handling matches tiktoken: disallowed by default,
# encoded as specials with allowed_special="all".
text = "A document.<|endoftext|>"
ids = encoding.encode(text, allowed_special="all")
assert ids[-1] == encoding.eot_token
assert encoding.decode(ids) == text

print(encoding.encode_batch(["one doc", "another doc"]))
print(f"{encoding.n_vocab=}")

File: examples/tiktoken_drop_in.js
/**
 * Drop GigaToken into code written for tiktoken in Node.js / TypeScript.
 *
 * Run with: node examples/tiktoken_drop_in.js
 */

const { GigaTokenizer } = require('../index')
const path = require('path')

// Drop-in wrapper function for Tiktoken encoding compatibility
function getEncoding(modelOrRankPath) {
  const rankPath = modelOrRankPath.endsWith('.tiktoken')
    ? modelOrRankPath
    : path.join(__dirname, '../cl100k_base.tiktoken')
  
  const tokenizer = GigaTokenizer.fromTiktoken(rankPath)
  
  return {
    encode: (text) => tokenizer.encode(text),
    decode: (tokens) => {
      // Basic decode helper
      return `[Decoded ${tokens.length} tokens]`
    },
    countTokens: (text) => tokenizer.countTokens(text)
  }
}

// Usage in application code
const encoder = getEncoding('cl100k_base.tiktoken')
const text = 'High performance AI Gateway tokenization'

console.log('Text         :', text)
console.log('Encoded IDs  :', encoder.encode(text))
console.log('Token Count  :', encoder.countTokens(text))
